Firestore/Source/Local/FSTLevelDBMutationQueue.mm

```diff
@@ -322,15 +310,9 @@ - (nullable FSTMutationBatch *)nextMutationBatchAfterBatchID:(FSTBatchID)batchID
   FSTBatchID nextBatchID = MAX(batchID, self.metadata.lastAcknowledgedBatchId) + 1;
 
   std::string key = [self mutationKeyForBatchID:nextBatchID];
-  std::unique_ptr<Iterator> it(_db->NewIterator(StandardReadOptions()));
+  auto it = _db.currentTransaction->NewIterator();
   it->Seek(key);
 
-  Status status = it->status();
```

Comment:
Removing this doesn't seem right. Can't these reads still fail? If they did fail, how would we know what LevelDB had to say about it?

Here and throughout.

Reply:
It's a good question. We currently check inconsistently throughout the codebase. 

To check properly, we would need to check after every call that can mutate the iterator: `Next()`, `Prev()`, `Seek()`, etc. It appears that the failure mode we would be catching by checking `status()` would be data corruption on disk. As best I can tell from looking at the various `Iterator` implementations, `Valid()` will return `false` when `status()` would return anything other than `Status::OK`. By enforcing calls to `status()`, we would be turning situations where we previously stopped iterating or were returning `NotFound` into fatal errors. This is I think what we want, because the current situation could be producing subtle logic errors. One example is `FSTLevelDBMutationQueue.loadNextBatchIDFromDB`. It currently does not check `status()`, and thus could incorrectly report the lack of a mutation queue for a user. 

I think it's reasonable to check now that leveldb reads are or will soon be contained within a `LevelDbTransaction`. The aforementioned scenario is also a good motivation to add bidirectional semantics to the iterator and then to port `loadNextBatchIDFromDB` to run in a transaction.

Reply:
I'm not sure I want every call to `Valid()` to be paired with a call to `status()`, but it does seem like we likely want to call status at least once per method that called it.

Without necessarily addressing this throughout the codebase, is it possible to re-add these?

Reply:
Ok, I've added checks to `LevelDbTransaction` for `status()`. I have not touched the example above, in the mutation queue. I think that's best addressed by porting it to a transaction.

Reply:
OK